prov/util/src/util_main.c

```diff
@@ -165,6 +165,44 @@ static int util_find_domain(struct dlist_entry *item, const void *arg)
 		 ((info->domain_attr->mr_mode & domain->mr_mode) == domain->mr_mode);
 }
 
+void util_lookup_existing_fabric_domain(const struct util_prov *util_prov,
+					const struct fi_info *hints,
+					struct fi_info **info)
+{
+	struct util_fabric *fabric;
+	struct util_domain *domain;
+	struct dlist_entry *item;
+	struct util_fabric_info fabric_info;
+
+	fabric_info.name = (hints && hints->fabric_attr && hints->fabric_attr->name) ?
+			    hints->fabric_attr->name : (*info)->fabric_attr->name;
```

Comment:
nit: The line length looks long.

Are you saying that the check for the tcp provider is wrong because the name stored in the opened instance differs from the more generic name assigned to an uninitialized, hard-coded fi_info being used as a base?

There are 2 changes being combined here.  Please apply the fix to the existing code (so it can be backported more easily), then move the code into a separate function.

If the check against the fabric name is wrong, is the check against the domain name also wrong for tcp?